File: src/test/java/com/github/ojacquemart/apiauchan/restaurant/menu/MenuTest.java
package com.github.ojacquemart.apiauchan.restaurant.menu;

import org.junit.Before;
import org.junit.Test;

import java.util.Arrays;

import static org.assertj.core.api.Assertions.assertThat;

public class MenuTest {

    private Menu menu;

    @Before
    public void setUp() throws Exception {
        menu = new Menu(Arrays.asList(
                new DishGroup("foo", Arrays.asList(new Dish("foo", "1€"), new Dish("bar", "2€"))),
                new DishGroup("bar", Arrays.asList(new Dish("bar", "3€"), new Dish("bar", "4€"))))
        );
    }

    @Test
    public void testToString() throws Exception {
        String menuAsString = menu.toString();

        assertThat(menuAsString).isNotNull();
        assertThat(menuAsString).isEqualTo(
            "API MENU\n\n" +
                    "foo\n" +
                    "\tfoo 1€\n" +
                    "\tbar 2€\n" +
                    "\n" +
                    "bar\n" +
                    "\tbar 3€\n" +
                    "\tbar 4€\n" +
                    "\n"
        );
    }
}

File: src/test/kotlin/com/github/ojacquemart/api/restaurant/menu/MenuTest.kt
package com.github.ojacquemart.api.restaurant.menu

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class MenuTest {

    private lateinit var menu: Menu

    @Before
    @Throws(Exception::class)
    fun setUp() {
        menu = Menu(listOf(
                DishGroup("foo", listOf(Dish("foo", "1€"), Dish("bar", "2€"))),
                DishGroup("bar", listOf(Dish("bar", "3€"), Dish("bar", "4€")))))
    }

    @Test
    fun testWrite() {
        val menuAsString = menu.write()

        assertThat(menuAsString).isNotNull()
        assertThat(menuAsString).isEqualTo(
                "*API MENU*\n\n" +
                        "> *foo*\n" +
                        "foo 1€\n" +
                        "bar 2€\n" +
                        "\n" +
                        "> *bar*\n" +
                        "bar 3€\n" +
                        "bar 4€\n" +
                        "\n")
    }

    @Test
    fun testWriteWithEmptyOptions() {
        val menuAsString = menu.write(emptyList())

        assertThat(menuAsString).isNotNull()
        assertThat(menuAsString).isEqualTo(
                        "> *foo*\n" +
                        "foo\n" +
                        "bar\n" +
                        "\n" +
                        "> *bar*\n" +
                        "bar\n" +
                        "bar\n" +
                        "\n")
    }

    @Test
    fun testWriteWithOnlyOneDishGroup() {
        val menu =  Menu(listOf(
                DishGroup("foo", listOf(Dish("foo", "1€"), Dish("bar", "2€")))))
        val menuAsString = menu.write(emptyList())

        assertThat(menuAsString).isNotNull()
        assertThat(menuAsString).isEqualTo(
                        "foo\n" +
                        "bar\n" +
                        "\n")
    }

}